### PR TITLE
Add warning about plugin permission

### LINKED ISSUE #4382: [DOC] Plugin installation page should describe the security posture of installed plugins
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 There should be text of some kind on the documentation site that helps OpenSearch users / operators understand the implications of what plugins can do

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by peternied**
### Description
I didn't see any documentation that described what an installed plugin can do - and it would be good to make sure cluster operators have expectations described somewhere.

### Issues Resolved
- Resolves #4382

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@cwillum Would love your feedback on this change

File: _install-and-configure/plugins.md [hunk 1]
```diff
@@ -11,6 +11,11 @@ redirect_from:
 
 You can install individual plugins for OpenSearch based on your needs. For information about available plugins, see [Available plugins](#available-plugins).
 
+## Considerations for Plugins
```
Comment (Naarcha-AWS) [inline _install-and-configure/plugins.md:14]:
```suggestion

```

File: _install-and-configure/plugins.md [hunk 2]
```diff
@@ -11,6 +11,11 @@ redirect_from:
...
+
+In OpenSearch Plugins have the ability to access the data in the cluster, including metadata about cluster operations. Take care when selecting and installing plugins - only use those from trusted sources. Ensuring the plugin's source is reliable helps to mitigate risks to cluster's data and integrity. Always vet the source and understand the function of a plugin before installing on an OpenSearch cluster.
```
Comment (Naarcha-AWS) [inline _install-and-configure/plugins.md:16]:
```suggestion
OpenSearch plugins give you the ability to access the data in the cluster, including metadata about cluster operations. To help mitigate risks to your cluster's data and integrity,  make sure that a custom plugin's source is reliable when selecting and installing custom OpenSearch plugins. Furthermore, make sure that you understand the function of a plugin before installing on an OpenSearch cluster.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Comment by peternied** [comment]
Thanks @Naarcha-AWS for the fast feedback and suggestions!

**Comment by peternied** [comment]
@Naarcha-AWS I've updated the text based on @cwillum's suggestions, let me know what you think

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**